source/adapters/level_zero/memory.cpp

```diff
@@ -2045,19 +2045,37 @@ ur_result_t _ur_buffer::getZeHandle(char *&ZeHandle, access_mode_t AccessMode,
 
   auto &Allocation = Allocations[Device];
 
+  if (this->isFreed) {
+    logger::debug("getZeHandle(ur_device_handle{}) buffer already released, no "
+                  "valid handles.",
+                  (void *)Device);
+    UR_ASSERT(!this->isFreed, UR_RESULT_ERROR_INVALID_NULL_HANDLE);
```

Comment:
`UR_ASSERT` is not a real assert: https://github.com/oneapi-src/unified-runtime/blob/d9608131d61159952db00bead44efffd3fab0ee3/source/adapters/level_zero/common.hpp#L306 We should get rid of that at some point, but in here I think we need to use [`die()`](https://github.com/oneapi-src/unified-runtime/blob/d9608131d61159952db00bead44efffd3fab0ee3/source/ur/ur.hpp#L63).

Reply:
Hello @pbalcer , I have updated the code to die() in these cases and I have added a missing refcnt during sub buffer creation such that this issue should be avoided in most cases, meaning the die() is now simply a guard.